### ISSUE TITLE
PF2E not measuring correctly on diagonals in regards to elevation

### ISSUE THREAD

**Issue by justjoekingg** [caewok/fvtt-elevation-ruler#227]
![measurement_dnd_5-10-5](https://github.com/user-attachments/assets/33265c22-3b77-41b3-a51c-4f0bf715556f)

Above is a screenshot from the github page for elevation ruler explaining/demonstrating how elevation ruler works in a 5-10-5 system such as pathfinder 2nd edition. It shows measuring 2 squares to the left and 10ft up should give you 15ft of total measurement.

In my game this does not happen, it seems like it's using dnd 5e measurement which is 5-5-5. Is there some kind of way to force it to use 5-10-5? Euclidean is disabled, its updated to the current version, I don't have any of the other modules from your list of modules that complicate this one or mess with it and have even gone so far as to create a fresh world with all other modules besides elevation ruler and libwrapper disabled, current up to date PF2E system as well, foundry version 12 stable build 331

**Comment by tnarv** [comment]
It seems changing elevation doesn't seem to respect any of the Square Diagonal settings. I tried all of them in a dnd5e world (in order to rule out PF2e specifically being a problem) and when changing elevation it always treated diagonals as costing a single grid square of movement regardless of setting.

**Comment by caewok** [comment]
Can you provide a screenshot of what you are trying to measure? When I try to measure diagonal movement, I see it using 5-10-5 if that setting is enabled.

**Comment by justjoekingg** [comment]
Normal diagonal movement works fine for me. As in just dragging diagonally it does 5-10-5 (two squares of diagonal movement equaling 15ft) but the issue I'm having is moving _vertically_ in a diagonal manner. In the image I posted up top that is from Elevation Ruler's main github screen. It shows what should be happening; 10ft forward and 10ft up is 15ft total movement, as that's essentially 2 diagonal movements, just measured vertically and diagonally.
But when you do that in foundry 10ft forward and 10ft up vertically is just displaying10ft, here's a picture from my game taken right now with PF2E updated and Elevation Ruler updated to current versions.

![Screenshot 2024-12-22 151117](https://github.com/user-attachments/assets/7e418948-873b-4082-b9ec-ebabef39b105)

For further proof that 10ft horizontal and 10ft vertical should be 15ft rather than 10, even a measurement shows they're 15ft away. 
![Screenshot 2024-12-22 151617](https://github.com/user-attachments/assets/f6068692-e869-41a5-b18b-ce050edc8591)

**Comment by caewok** [comment]
Yep, the calculation for diagonals was counting a straight + vertical as straight instead of diagonal. Addressed in v0.10.17.

 I removed the description about diagonals that you quoted above from the README because I don't think it is strictly accurate anymore. I have since moved toward what Foundry v13 appears to be doing for these calculations, so some of it may have changed.

Keep in mind that there are 2 types of diagonals in 3d. The 2d diagonal of a 1x1 square should be `sqrt(2)` while the 3d diagonal of a 1x1x1 cube should be `sqrt(3)` for euclidean measurements. Moving horizontally + elevating, which is what I think you were describing above, would be `sqrt(2)`. Approximate measurement uses `1.5` and `1.75`; rectilinear uses `2` and `3`.  Obviously equidistance uses `1` for both. 5|10|5 treats both diagonal types as one diagonal move.
